packages/manager/apps/web-domains/src/alldoms/hooks/useAllDomDatagridColumns.tsx

```diff
@@ -70,8 +71,9 @@ export const useAllDomDatagridColumns = (
       {
         id: 'nicAdmin',
         cell: (props: TServiceDetail) => {
-          const contact = props.serviceInfo.customer.contacts.find(
-            (c) => c.type === ServiceInfoContactEnum.Administrator,
+          const contact = findContact(
```

Comment:
same here you are searching for the contact on every render, if you would have it already included in your `TServiceDetail` you would do it only once, when you get the API response